src/hotspot/share/gc/shared/gcVMOperations.cpp

```diff
@@ -167,7 +167,16 @@ void VM_GC_HeapInspection::doit() {
     }
   }
   HeapInspection inspect;
-  inspect.heap_inspection(_out, _parallel_thread_num);
+  WorkerThreads* workers = Universe::heap()->safepoint_workers();
+  if (workers != nullptr) {
+    // The GC provided a WorkerThreads to be used during a safepoint.
+    // Can't run with more threads than provided by the WorkerThreads.
+    const uint capped_parallel_thread_num = MIN2(_parallel_thread_num, workers->max_workers());
+    WithActiveWorkers with_active_workers(workers, capped_parallel_thread_num);
```

Comment:
I would *almost* rely on the workgang's 'active_workers()' here because all collectors set this proportional to heap size (people tend to start 128M VMs on hundreds-of-thread machines... ). The problem is that one heap inspection VM operation. Maybe put the `WithActiveWorkers` there and use `workers->active_workers()` here?

This is just a suggestion.

Reply:
Sorry, I don't think I understand the suggestion. Are you suggesting to use WithActiveWorkers only for the heap inspection VM op, and workers->active_workers() everywhere else (for the collector use-case)?

> The problem is that one heap inspection VM operation. Maybe put the WithActiveWorkers there

This area *is* that one heap inspection VM operation, isn't it?

Reply:
You are right, I misread the code. Ignore my comment :) Ship it.

Reply:
Cool, thanks for reviewing :)